base/gcutils.jl

```diff
@@ -81,6 +89,92 @@ mutable struct MyMutableStruct
     end
 end
 ```
+
+# Extended help
```

Comment:
While you're here, can you comment on why `GC.disable_finalizers()` is required, given that `lock(my_lock)` will disable finalizers once it acquires the lock (for the locks included in Base, and presumably also for any correctly implemented locks in packages)?

---

Re: where to put the info:

The docstrings are much more visible at the REPL and in editors with LSP help previews, so if the info is important then it seems like a good idea to make it easy to find?

If we could put a hyperlink in to the relevant manual section then that would be fine, but the REPL obviously doesn't support those.

Anyway, I have a slight preference, but I can move the content back into the manual if you or other reviewers have stronger preferences for that way.

---

Re: strategy 2, sure, removed. Distributed.jl still uses basically that strategy, tho. Is that a problem?

```julia
function finalize_ref(r::AbstractRemoteRef)
    if r.where > 0 # Handle the case of the finalizer having been called manually
        if trylock(client_refs.lock) # trylock doesn't call wait which causes yields
            try
                delete!(client_refs.ht, r) # direct removal avoiding locks
                if isa(r, RemoteChannel)
                    send_del_client_no_lock(r)
                else
                    # send_del_client only if the reference has not been set
                    v_cache = @atomic :monotonic r.v
                    v_cache === nothing && send_del_client_no_lock(r)
                    @atomic :monotonic r.v = nothing
                end
                r.where = 0
            finally
                unlock(client_refs.lock)
            end
        else
            finalizer(finalize_ref, r)
            return nothing
        end
    end
    nothing
end
```

Reply:
True, strategy 1 seems to already be implicit in the use of locks (and is mentioned as such), which makes it an odd case to describe as such, instead of just pointing out that locks are always needed